for_developers/dir_structure.md

```diff
@@ -0,0 +1,38 @@
+```console
+root/
+    algo/                       # Custom algo (e.g., hierarchical_cls_head)
+    cli/                        # CLI entrypoints
+    config/                     # Default YAML config files
+    core/
+        config/                 # Structured data type object for configurations
+        data/                   # Data related things
+            dataset/            # OTXDataset
+                base.py
+                detection.py
+                ...
+            entity/             # OTXDataEntity
+                base.py
+                detection.py
+                ...
+            transform_libs/     # To support transform libraries (e.g., MMCV)
+            factory.py          # Factory to instantiate data related objects
+            module.py           # OTXDataModule
+        engine/                 # PyTorchLightning engine
+            train.py
+            ...
+        model/                  # Model related things
+            entity/             # OTXModel
+                base.py
+                detection.py
+                ...
+            module/             # OTXLitModule
+                base.py
+                detection.py
+                ...
+        types/                  # Enum definitions (e.g. OTXTaskType)
+        utils/                  # Utility functions
+    recipe/                     # Recipe YAML config for each model we support
```

Comment:
What distinguishes the `recipe` from the `config`? Isn't a `recipe` essentially a type of configuration? Would it be more intuitive to nest it under `config` and assign a more descriptive name? The term `recipe` isn't commonly used in the DL field, which could potentially cause confusion.

Reply:
It is a collection of YAML configuration file for each model. It is the final one will override other default configurations in `/config` at the last moment. The reason why it is in the separate directory is that we can `python cli/train.py +recipe=detection/rtmdet_tiny.py` with using it as the final overriding configuration. It is referred from https://hydra.cc/docs/patterns/configuring_experiments/.

I don't know that "recipe" sounds weird in the perspective of the English native. I don't have any objection for you new suggestion about this since you are native in English. But, what is the difference in nuisance between the `recipe` directory in this PR and this directory you guys has been maintaining until today?
https://github.com/openvinotoolkit/training_extensions/blob/develop/src/otx/recipes/__init__.py#L1
I'm just curious about it.